backend/main.py

```
@@ -562,6 +596,33 @@ async def get_project_tasks(project_id: str):
+        task = await task_service.update_task(project_id, task_id, updates)
+        if not task:
+            logger.warning(f"Task not found: {task_id}")
+            raise HTTPException(status_code=404, detail="Task not found")
+
+        logger.info(f"Task marked completed successfully: {task_id}")
+        return task
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error completing task {task_id}: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to complete task: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)